src/main/java/seedu/duke/controllers/DishController.java

```diff
@@ -67,54 +67,82 @@ private void listDishes() {
+        if (name.isEmpty()) {
+            System.out.println("Please make sure the name is not empty and price is non-negative");
+            return;
+        }
+        if (dishManager.isNameExist(name)) {
+            System.out.println("Ooops, this dish already exists in your menu");
+            return;
         }
+        dishManager.setName(index, name);
     }
 
     private void changePrice() throws OperationTerminationException {
+        if (dishManager.getNumOfDishes() == 0) {
+            System.out.println("You haven't added a dish!");
+            return;
+        }
         MainLogger.logInfo(this, "User is changing the price of dish");
         System.out.println("Changing price...");
         int index = InputParser.getInteger("The index of dish: ");
-        double newPrice = InputParser.getDouble("The new price of dish: ");
-        try {
-            dishManager.setPrice(index, newPrice);
-        } catch (IndexOutOfBoundsException e) {
+        if (index <= 0 || index > dishManager.getNumOfDishes()) {
             System.out.println("Please make sure the index is valid");
-        } catch (IllegalArgumentException e) {
+            return;
+        }
+        double newPrice = InputParser.getDouble("The new price (will be rounded into 1 decimal place) of dish: ");
+        if (newPrice < 0) {
             System.out.println("Please make sure the price is not negative");
+            return;
         }
+        dishManager.setPrice(index, newPrice);
     }
 
     private void deleteDish() throws OperationTerminationException {
+        if (dishManager.getNumOfDishes() == 0) {
+            System.out.println("You haven't added a dish!");
+            return;
+        }
         MainLogger.logInfo(this, "User is deleting dish");
         System.out.println("Deleting dish...");
         int index = InputParser.getInteger("The index of dish: ");
-        try {
-            dishManager.deleteDish(index);
-        } catch (IndexOutOfBoundsException e) {
+        if (index <= 0 || index > dishManager.getNumOfDishes()) {
             System.out.println("Please make sure the index is valid");
+            return;
         }
+        dishManager.deleteDish(index);
     }
 
     private void addDish() throws OperationTerminationException {
         MainLogger.logInfo(this, "User is adding dish");
         System.out.println("Adding new dish...");
         String name = InputParser.getString("The name of dish: ");
-        double price = InputParser.getDouble("The price of dish: ");
-        try {
-            dishManager.addDish(new Dish(name, price));
-        } catch (IllegalArgumentException e) {
-            System.out.println("Please make sure the name is not empty and price is positive");
+        if (name.isEmpty()) {
+            System.out.println("Please make sure the name is not empty and price is non-negative");
+            return;
+        }
+        if (dishManager.isNameExist(name)) {
```

Comment:
I think "is..." is only for naming boolean attributes, correct me if I am wrong though!

Reply:
You are correct!
I renamed it already.